src/hooks/core/chat/useDMMessage.ts

```diff
@@ -0,0 +1,112 @@
+import {v4 as uuid} from 'uuid';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import ChannelListMemberSchema from '../../../database/schema/ChannelListMemberSchema';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import ChatSchema from '../../../database/schema/ChatSchema';
+import UserSchema from '../../../database/schema/UserSchema';
+import {DEFAULT_PROFILE_PIC_PATH} from '../../../utils/constants';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import useChatUtilsHook from './useChatUtilsHook';
+import {initChatFromPost, initChatFromPostAnon} from '../../../service/chat';
+
+type ChannelCategory = 'SIGNED' | 'ANONYMOUS';
+
+const useDMMessage = () => {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
+  const {goToChatScreen} = useChatUtilsHook();
+
+  const saveChannelList = async (channel, type: ChannelCategory) => {
+    try {
+      if (channel?.members?.length === 0) return Promise.reject(Error('no members'));
+      let userTarget = channel?.better_channel_members.find(
+        (data) => data?.user?.id !== signedProfileId
+      );
+
+      if (type === 'ANONYMOUS') {
+        userTarget = channel?.better_channel_members.find(
+          (data) => data?.user?.id !== anonProfileId
+        );
+      }
+
+      const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
+      const chatName = {
+        name: userTarget?.user?.name || userTarget?.user?.username,
+        image: userTarget?.user?.image || DEFAULT_PROFILE_PIC_PATH
+      };
+
+      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      channel.myUserId = signedProfileId; // change to use getChannelListInfo
+      channel.targetName = chatName?.name; // change to use getChannelListInfo
+      channel.targetImage = chatName?.image; // change to use getChannelListInfo
+
+      channel.channel = {...channel};
+      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members);
+      await channelList.saveIfLatest(localDb);
+      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
+      return goToChatScreen(channelList);
+    } catch (error) {
+      console.log('sendMessage', error);
+    }
+  };
+
+  const sendMessageDM = async (
+    id,
+    source: 'post' | 'comment',
+    channelCategory: ChannelCategory
+  ) => {
+    const initChat =
+      channelCategory === 'ANONYMOUS'
+        ? await initChatFromPostAnon({source, id})
+        : await initChatFromPost({source, id});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChat` assignment could be simplified by using a ternary operator inside the function call.

```diff
-     const initChat =
-       channelCategory === 'ANONYMOUS'
-         ? await initChatFromPostAnon({source, id})
-         : await initChatFromPost({source, id});
+     const initChat = await (channelCategory === 'ANONYMOUS' 
+         ? initChatFromPostAnon({source, id}) 
+         : initChatFromPost({source, id}));
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->